### PR TITLE
fix: fork template button not visible on page refresh

### PR DESCRIPTION
**Pull request by akash-codemonk**
## Description

The same component is used for both forking templates to a workspace and fork a template to an existing app. For forking a template to a workspace, the fork button is visible only if there is a workspace visible to fork to. In the editor we don't fetch the workspace list api so the workspace list is empty by default when we load the appsmith app starting from the editor route. The fix here is to show the for button always inside the editor. 

Also fixes other misc issues  
- Clicking on fork button was calling the fork api twice.
- Clicking on the fork button was also triggering the click on the parent div(template card view) which then called template details api.
- Updated cypress tests for the same.

### TLDR
Fixes an issue with the fork button not being visible in the editor when the page is refreshed.

Fixes https://github.com/appsmithorg/appsmith/issues/20387
Fixes https://github.com/appsmithorg/appsmith/issues/17275

### Media
on release

https://user-images.githubusercontent.com/67054171/216881837-3c9e1e00-fca5-402e-8b9b-ec0ded868175.mov

on deploy preview

https://user-images.githubusercontent.com/67054171/216881923-af576a5d-ac0c-4dcc-b53b-f156227b2844.mov

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Manual
- Cypress

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Templates/ForkTemplateToGitConnectedApp.js [hunk 1]
```diff
@@ -32,11 +32,6 @@ describe("Fork a template to the current app", () => {
     );
     cy.wait(1000);
     cy.get(template.templateDialogBox).should("be.visible");
-    cy.wait("@getTemplatePages").should(
```
Comment (akash-codemonk) [inline app/client/cypress/integration/Regression_TestSuite/ClientSideTests/Templates/ForkTemplateToGitConnectedApp.js:35]: Previously the template pages api was being called to get the list of pages in a template while forking a template, this is no longer called as it isn't required.

File: app/client/cypress/support/commands.js [hunk 2]
```diff
@@ -1016,7 +1016,9 @@ Cypress.Commands.add("startServerAndRoutes", () => {
   cy.intercept("POST", "/api/v1/datasources/mocks").as("getMockDb");
   cy.intercept("GET", "/api/v1/app-templates").as("fetchTemplate");
   cy.intercept("POST", "/api/v1/app-templates/*").as("importTemplate");
-  cy.intercept("GET", "/api/v1/app-templates/*").as("getTemplatePages");
+  cy.intercept("GET", /\/api\/v1\/app-templates\/(?!(filters)).*/).as(
```
Comment (akash-codemonk) [inline app/client/cypress/support/commands.js:1019]: We want to intercept /api/v1/app-templates/* accept /api/v1/app-templates/filters. There are two different apis `app-templates/<template-id>` and `app-templates/filters`

File: app/client/src/pages/Templates/Template/index.tsx [hunk 3]
```diff
@@ -143,6 +138,7 @@ export function TemplateLayout(props: TemplateLayoutProps) {
   const onForkButtonTrigger = (e: React.MouseEvent<HTMLElement>) => {
     if (props.onForkTemplateClick) {
       e.preventDefault();
+      e.stopPropagation();
```
Comment (akash-codemonk) [inline app/client/src/pages/Templates/Template/index.tsx:141]: Click on a child element was calling the onClick handler of the wrapper/parent element, so stopping the propagation here.

File: app/client/src/pages/Templates/Template/index.tsx [hunk 4]
```diff
@@ -156,55 +152,54 @@ export function TemplateLayout(props: TemplateLayoutProps) {
...
-                    icon="plus"
-                    onClick={onForkButtonTrigger}
-                    size={Size.medium}
-                    tag="button"
+    <>
+      <ForkTemplateDialog
+        onClose={onForkModalClose}
+        showForkModal={showForkModal}
+        templateId={id}
+      />
```
Comment (akash-codemonk) [inline app/client/src/pages/Templates/Template/index.tsx:160]: Moved out the blueprint dialog component to not have it wrap of the child element with the click event handler. stopPropagation added above had no effect with this wrapped the element.

**Review by hetunandu — APPROVED** [review]
